### DISCUSSION TITLE
Supporting videos / animated GIFs

### DISCUSSION THREAD

**Discussion by awestlake87** [emilk/egui#1606] [Q&A] [answered]
Hey, I'm looking at using this library for one of my hobby projects, but I'll probably need some support for viewing / scrubbing through video files. I saw some existing support for images, but wasn't able to find any examples for videos.

Does anyone have any examples of using videos with egui?

Ideally, I'd like to stick to the Windows desktop environment since I want to bind to some Python libraries. I'm not completely opposed to web-based solutions for this, but at that point it might make more sense for me to just continue on with something like PyQt5.

**Comment by coderedart** [comment, marked as answer]
afaik, there's no video player functionality. 

you are better off launching a separate video player app to handle the playback (like mpv). mpv can also be embedded into your own window using libmpv https://github.com/mpv-player/mpv-examples/blob/master/libmpv/README.md
EDIT: mpv uses opengl too. 

if you are really really interested in making your own media player, then you can try using gstreamer https://crates.io/crates/gstreamer-player .

I would honestly recommend just going with qt5 though.

**Comment by aadityaverma** [comment]
Update : https://github.com/n00kii/egui-video

https://crates.io/crates/egui-video/0.6.0/dependencies
